rep-2002.rst

```diff
@@ -0,0 +1,137 @@
+
+
+Motivation
+==========
+
+Create a rolling ROS distribution to serve as the foundation for subsequent stable ROS distributions in order to reduce the workload of Open Robotics staff and individual package maintainers when bootstrapping new ROS releases.
+
+
+Support status
+==============
+
+The rolling release itself will not be considered a supported platform.
+Distribution syncs will be performed on a regular cadence just like stable distributions but regressions may not block a rolling release sync.
+During the pre-release phase for stable ROS distributions releases in the rolling distribution may be blocked or held back in order to conform to the support requirements of the upcoming stable release.
+
+
+Release process
+===============
+
+Releasing stable ROS distributions from a rolling release
+---------------------------------------------------------
+
+When a new stable release is being prepared, the rolling release will be used as a staging ground to prepare that release.
+Updates and new package releases made to the rolling release will be propagated automatically to the stable release.
+At an announced time prior to the official stable release, the stable distribution will be created and packages with successful builds on the rolling release will be automatically released via bloom into the stable release.
+Once the stable distribution has been created and the automatic bloom process has been run, changes to the rolling release may not be automatically propagated to the stable release.
+If a package is prevented from being automatically released into the stable distribution due to a dependency being unavailable in the stable distribution, that package may be automatically re-bloomed when the dependency becomes available in the stable distribution.
+
+
+Updating the rolling release platforms
+--------------------------------------
+
+The rolling release is intended to remain availabe indefinitely and so will not be retired or reach end-of-support status without an update to this REP and the ROS distribution release process.
+However, this does mean that platform changes during the life of the rolling distribution are going to occur and the rolling release may target platforms that are themselves unstable or in a pre-release state.
+When a platform change is coming, the intended date of change will be announced ahead of time.
+Once the platforms have been updated, everything currently in the rolling release will be automatically re-released via bloom to target the updated release platforms.
+
+
+Bootstrapping the rolling release
+---------------------------------
+
+The first release of the rolling rosdistro will use the same procedure we've used for all stable rosdistros previously.
+Packages will be released via bloom into an empty distribution until the distribution is relatively complete.
+Package maintainers may then choose to make releases into the rolling rosdistro.
```

Comment:
Just a clarification here, does this mean that there will be 2 steps?

1. _Packages will be released via bloom into an empty distribution_
1. _Package maintainers may then choose to make releases into the rolling rosdistro_

Couldn't the "empty distribution" be the "rolling distribution", requiring only one action from maintainers?

Reply:
I'll reword this.

I did write it as two steps:
1. Bootstrap the rolling distro. Which will begin as all distributions do, empty.
2. Use the rolling distro as a possible release track available for maintainers.

They won't be releasing the same version multiple times, the second step there is for continuing releases.